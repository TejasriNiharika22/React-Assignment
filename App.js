import React from 'react';
import './App.css';
import AddTodo from './Common/SeparateComponent/AddTodo';
import TodoList from './Common/SeparateComponent/TodoList';
import Demo from './Demo/MockFile';
class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      data: []
    };
  }

  addItem(item) {
    this.state.data.unshift({
      task: item
    });
    this.setState({
      data: this.state.data
    });
  }

  findItem(item) {
    // return this.state.data.filter((element) => element.task === item)[0]; 
    return this.state.data.find((element) => element.task === item) 
  }

  saveItem(oldItem, newItem) {
    let selectedItem = this.findItem(oldItem);
    selectedItem.task = newItem;
    this.setState({ data: this.state.data });
  }

  deleteItem(item) {
    let index = this.state.data.map(element => element.task).indexOf(item);
    this.state.data.splice(index, 1);
    this.setState({ data: this.state.data });
  }
  render() {
    return (
      <div className="App">
        <div className="header">
          <h1>{this.props.Content.text}</h1>
          <br />
        </div>
        <AddTodo data={this.state.data} addItem={this.addItem.bind(this)} />
        <br />
        <br />
        <h2>{this.props.Content.sub}</h2>
    
        <TodoList data={this.state.data} deleteItem={this.deleteItem.bind(this)} saveItem={this.saveItem.bind(this)}
        />
      </div>
    );
  }
}
App.defaultProps = {
  Content: Demo
};

export default App;