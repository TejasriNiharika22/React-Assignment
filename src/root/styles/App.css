.App {
  text-align: center;
  min-height: 500px;
  width: 400px;
  margin: 40px auto;
}

h1 {
  text-align: center;
  color: cadetblue;
  /* background: transparent; */
  text-shadow: 0 2px 2px black;
  /* background-color: whitesmoke; */
  /* padding: 6px 2px 3px 1px;
  box-shadow: 5px 5px 15px -5px rgba(0, 0, 0, 0.3); */
}

h2 {
  color: black;
  text-shadow: hotpink;
}

input {
  min-height: 50px;
  min-width: 50px;
  text-align: center;
  height: 50px;
  top: 10px;
  border: none;
  background: transparent;
  font-size: 20px;
  font-weight: 500;
  width: 20%;
  align-items: center;
  border-bottom: 0.5px solid #8566;
  border-top: 0.5px solid #8566;
  border-left: 0.5px solid#8566;
  border-right: 0.5px solid #8566;
  
}

Button {
  align-items: center;
  min-height: 50px;
  width: 80px;
  border-radius: 8%;
  border-color: transparent;
  background-color: #008cba;
  color: #ffff;
  font-size: 20px;
  border: none;
  outline: none;
  margin-left: 10px;
  box-shadow: 5px 5px 15px -5px rgba(0, 0, 0, 0.3);
  text-align: center;
}

.item {

  width: 350px;
  padding-left: 0px;
  text-align: center;
  font-size: 20px;
  font-weight: 500;
  min-height: 40px;
  display: flex;
  align-items: center;
  color: #8566aa;
  text-transform: capitalize;
  border-bottom: 2px solid grey;
  border-top: 2px solid grey; 
  border-left: 2px solid grey;
  border-right: 2px solid grey;
  border-radius: 8px;
}

.editdel{
  float: right;
  text-align: right;
  padding-right: 5px;
  nav-right: auto; 
}

.actions{
text-align: right;
align-items: right;

}
