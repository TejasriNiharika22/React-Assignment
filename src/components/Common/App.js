import React from 'react';
import styles from '../../root/styles/App.css';
import AddTodo from '../../components/Common/AddTodo'
import TodoList from '../../components/Common/TodoList'
import data from '../../mock/MockData'
class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: []
    };
  }
  addItem(item) {
    this.state.data.unshift({
      task: item
    });
    this.setState({
      data: this.state.data
    });
  }
  findItem(item) {
    // return this.state.data.filter((element) => element.task === item)[0]; 
    return this.state.data.find((element) => element.task === item) 
  }
  saveItem(oldItem, newItem) {
    let selectedItem = this.findItem(oldItem);
    selectedItem.task = newItem;
    this.setState({ data: this.state.data });
  }
  deleteItem(item) {
    let index = this.state.data.map(element => element.task).indexOf(item);
    this.state.data.splice(index, 1);
    this.setState({ data: this.state.data });
  }
  render() {
    return (
      <div className= {styles.App}>
        <div className="header">
          <h1>{this.props.Content.text}</h1>
          <br />
        </div>
        <AddTodo data={this.state.data} addItem={this.addItem.bind(this)} />
        <br />
        <br />
        <h2>{this.props.Content.sub}</h2>
    
        <TodoList data={this.state.data} deleteItem={this.deleteItem.bind(this)} saveItem={this.saveItem.bind(this)}
        />
      </div>
    );
  }
}
App.defaultProps = {
  Content: data
};
export default App;