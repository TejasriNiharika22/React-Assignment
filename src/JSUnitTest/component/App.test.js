import { shallow } from 'enzyme';
import App from '../../components/App'

describe('rendering my component', () => {
  let props = {
    task: "",
    deleteItem: fn => fn,
    saveItem: fn => fn,
    findItem: fn => fn
  }
  it('should test App component with default state of empty array', () => {
    const wrapper = shallow(<App data={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should call addItem method', () => {
    const value = {
      target: {
        item: "Create new Item"
      }
    }
    const wrapper = shallow(<App />);
    wrapper.instance().addItem(value);
    wrapper.setState({
      data: { task: "task1" }
    })
    expect(wrapper.state('data')).toEqual({ task: "task1"});
  });
  it('should call saveItem method', () => {
    const oldItem =  ['task1', 'task2']
    const newItem = [{ selectedItem: "task1added" }, "task2added"]
    const wrapper = shallow(<App {...props} />);
    wrapper.setState({
      data: [{ task: "Selected Item" }]
    })
    wrapper.instance().saveItem(oldItem, newItem);
    expect(wrapper.state('data')).toEqual([{ task: "Selected Item" }]);
  });

  it('should call deleteItem method', () => {
    const item = ["Selected item","Another selected item"]
    const wrapper = shallow(<App {...props} />);
    wrapper.setState({
      data:[{task:"Selected Item"}]
    })
     wrapper.instance().deleteItem(item);
     expect(wrapper.state('data')).toEqual([]);
  });
});