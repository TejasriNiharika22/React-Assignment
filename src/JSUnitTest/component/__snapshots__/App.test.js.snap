// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering my component should test App component with default state of empty array 1`] = `ShallowWrapper {}`;
